accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/ImpersonationInstallerServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * (C) Copyright 2017 Netcentric AG.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package biz.netcentric.cq.tools.actool.authorizableinstaller.impl;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.jcr.RepositoryException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jackrabbit.api.security.principal.PrincipalIterator;
+import org.apache.jackrabbit.api.security.user.Impersonation;
+import org.apache.jackrabbit.api.security.user.User;
+import org.apache.jackrabbit.oak.spi.security.principal.PrincipalImpl;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import biz.netcentric.cq.tools.actool.configmodel.AuthorizableConfigBean;
+import biz.netcentric.cq.tools.actool.configmodel.AuthorizablesConfig;
+import biz.netcentric.cq.tools.actool.history.InstallationLogger;
+
+/** Installs impersonators to users. */
+@Component(service=ImpersonationInstallerServiceImpl.class)
+public class ImpersonationInstallerServiceImpl {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ImpersonationInstallerServiceImpl.class);
+
+    public void setupImpersonation(User user, List<String> impersonationAllowedFor, AuthorizablesConfig authorizablesConfig, InstallationLogger installationLog) {
+
+        
+        try {
+            List<String> impersonatorsToBeSet = expandImpersonatorsList(user.getID(), impersonationAllowedFor, authorizablesConfig);
+            Impersonation impersonation = user.getImpersonation();
+            List<String> currentImpersonators = collectCurrentImpersonators(impersonation);
+            applyChanges(user, installationLog, impersonatorsToBeSet, impersonation, currentImpersonators);
+        } catch (RepositoryException e) {
+            throw new IllegalStateException("Could not setup impersonation for user "+user + ": "+e, e);
+        }
+    }
+
+    private void applyChanges(User user, InstallationLogger installationLog, List<String> impersonatorsToBeSet, Impersonation impersonation,
+            List<String> currentImpersonators) throws RepositoryException {
+        List<String> impersonatorsToAdd = new ArrayList<>();
+        impersonatorsToAdd.addAll(impersonatorsToBeSet);
+        impersonatorsToAdd.removeAll(currentImpersonators);
+        Iterator<String> impersonatorsToAddIt = impersonatorsToAdd.iterator();
+        while(impersonatorsToAddIt.hasNext()) {
```

Comment:
some whitespace missing, are you using the right formatter?

Reply:
didn't have formatting on save turned on, fixed in 2e37da80c88ad3cf677c8025834df868d282e1a5